### ISSUE TITLE
Feedback testing debian packages: 9.3.4-rcx

### ISSUE THREAD

**Issue by t-b** [tango-controls/TangoSourceDistribution#67]
Notes: All java programs were used from the current source distribution from [here](https://github.com/tango-controls/TangoSourceDistribution/releases/tag/9.3.4-rc2).

Result:

- Package upgrade worked to `9.3.4~rc2+dfsg2-1~exp3` from the current unstable package

```
New Version: 9.3.4~rc2+dfsg2-1~exp3
Old Version: 9.2.5a+dfsg1-2+b2
```
- Jive worked with TangoTest and some of my tango device servers
- Astor worked but showed heartbeat error, see [here](https://github.com/tango-controls/Astor/issues/18)
- Pogo does not work see [here](https://github.com/tango-controls/pogo/issues/83)

- Database upgrade did not work. A reinstall of tango-db also did not create the correct database. I've checked with

```
MariaDB [tango]> SELECT * FROM property_class WHERE value ='EventConfirmSubscription';
Empty set (0.001 sec)
```

as that database entry does only exist in 9.3.4~rc2. TangoTest works though.

Doing a full reinstall and upgrade via

```
apt remove --purge "tango*"
apt install tango-db // answering questions regarding automatic management with db-config this yes
apt install -t experimental tango-db // dito
```

did not change anything.

I'm using `mariadb-serverdb`. I' attaching a database dump [tango-db.tango.2020-01-21-21.13.mysql.guCsZo.zip](https://github.com/tango-controls/TangoSourceDistribution/files/4093517/tango-db.tango.2020-01-21-21.13.mysql.guCsZo.zip)
 which was created by setting `export dbc_debug=1` and uninstalling `tango-db`.

Questions/Suggestions/Bugs:
- ~~Missing dbg package for libtango9~~ Found it, this is called libtango9-dbgsym and needs the following sources.list `http://debug.mirrors.debian.org/debian-debug experimental-debug/main amd64 Packages`.
- I think libtango9 should have a versioned dependency on tango-common because it is easily forgotten to upgrade if doing the package upgrade manually from experimental.
- ~~Missing libtango9-doc package~~ It is named libtango-doc and works nicely!
- ~~I had a hard time finding tango_admin, it is in the package libtango-tools which is okay (but nevertheless the package documentation shown with apt show libtango-tools should mention tango_admin), but the file itself should not be in a lib directory but better in bin. The tango device servers are good in /usr/lib/tango.~~ Created https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=949540.
- It would be nice to have a tango-full package to install everything cpptango related. In that way we can just tell users to install tango-full instead of `tango-db libtango9 ...`.

As part of testing the following Issues/PRs have been created:

- https://github.com/tango-controls/pogo/issues/83
- https://github.com/tango-controls/Astor/issues/18

- https://github.com/tango-controls/TangoSourceDistribution/issues/63
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=948496
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=949540

Done:
- https://github.com/tango-controls/TangoSourceDistribution/issues/62
- https://github.com/tango-controls/TangoSourceDistribution/pull/64
- https://github.com/tango-controls/TangoSourceDistribution/pull/65
- https://github.com/tango-controls/TangoSourceDistribution/pull/66

@bourtemb @picca

**Comment by picca** [comment]
Hello, thanks for this test

> Result:

>  *   Package upgrade worked to 9.3.4~rc2+dfsg2-1~exp3

Can you be more specific, because you explain that the upgrade of the database does not work ?


>  *   Jive worked with TangoTest and some of my tango device servers

I did not yet packages the java part. so whcih one are you using ?

>  *   Astor worked but showed heartbeat error, see here<https://github.com/tango-controls/Astor/issues/18>

idem

>  *   Pogo does not work see here<https://github.com/tango-controls/pogo/issues/83>

idem

>  *   Database upgrade did not work. A reinstall of tango-db also did not create the correct database. I've checked with

> MariaDB [tango]> SELECT * FROM property_class WHERE value ='EventConfirmSubscription';
> Empty set (0.001 sec)

> as that is only supposed to exist in 9.3.4. TangoTest works though.

I do not understant this last sentance. I packaged the 9.3.4-rc2 so it should work.
Is it possible to see the error message


> Questions/Suggestions/Bugs:

>  *   Missing dbg package for libtango9

This is normal https://wiki.debian.org/DebugPackage
now the debug symbols are automatically generated.

>   *   I think libtango9 should have a versioned dependency on tango-common

Can you explain why ?

>  *   Missing libtango9-doc package

In fact the -doc is in libtango-doc.
Since I support only one -dev version installed.

If we want to have different verion of tango-dev installed this is another story :))

>   *   I had a hard time finding tango_admin, it is in the package libtango-tools which is okay (but nevertheless the package documentation shown with apt show > libtango-tools should mention tango_admin), but the file itself should not be in a lib directory but better in bin. The tango device servers are good in /usr/lib/tango.

can you fill a bug against tango in Debian ?

>  *   It would be nice to have a tango-full package to install everything tango related

Can you explain what should be in this package. For now we only have tango C++ library the (db, starter, accesscontrol test devices), taurus, sardana, avaiulable in Debian. (I do not speak about the java part since it is not possible for now to build it and distribut it in the main distribution.

Cheers


Frederic

**Comment by mliszcz** [comment]
Hi, just one question:
> I had a hard time finding tango_admin, it is in the package libtango-tools which is okay (but nevertheless the package documentation shown with apt show libtango-tools should mention tango_admin), but the file itself should not be in a lib directory but better in bin. The tango device servers are good in /usr/lib/tango.

What is the motivation behind putting executable applications inside lib? From end-user perspective, I find this quite surprising (I expect apps to be installed to bin, which is usually on system search path).

**Comment by picca** [comment]
> Hi, just one question:

> I had a hard time finding tango_admin, it is in the package libtango-tools which is okay (but nevertheless the package documentation shown with apt show libtango-tools > should mention tango_admin), but the file itself should not be in a lib directory but better in bin. The tango device servers are good in /usr/lib/tango.

> What is the motivation behind putting executable applications inside lib? From end-user perspective, I find this quite surprising (I expect apps to be installed to bin, which is usually on system search path).

Hello, Debian maintainer speaking here, I decided to put all devices servers under /usr/lib/tango in order to avoid polluting the PATH with devices which should be managed by the starter device.
A bit like all the git command line, where only the git command is in the PATH.

For tango_admin, I agreed that it could be in /usr/bin instead of /ust/lib/tango.

cheers

Fred

**Comment by t-b** [comment]
@picca I've updated my post and did some more testing.

Regarding the java part of tango. AFAIU there is a licensing issue right? Would we be able to create non-free packages?

**Comment by picca** [comment]
> @picca<https://github.com/picca> I've updated my post and did some more testing.

thanks for the bug report :).

I will try to find some time in order to check your post.

> Regarding the java part of tango. AFAIU there is a licensing issue right? Would we be able to create non-free packages?

I usually do this, but this time I thnk that there is an issue with java-11 and the removal of CORBA from the jdk...

This should be fixed in the .jar provided in the tango source.

Cheers

Fred

**Comment by t-b** [comment]
@picca The java packages as they are depend on java-8.

**Comment by picca** [comment]
> @picca<https://github.com/picca> The java packages as they are depend on java-8.

And this is the problem. On Debian Buster (aka 10), there is only java-11.

**Comment by t-b** [comment]
@picca A new version is out, see https://github.com/tango-controls/TangoSourceDistribution/releases/tag/9.3.4-rc4.

**Comment by picca** [comment]
uploaded https://tracker.debian.org/news/1140670/accepted-tango-934rc4dfsg1-1exp1-source-into-experimental/

**Comment by t-b** [comment]
@picca 
I did some more testing with rc6:

- [x] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=949549 is not yet fixed. I'm seeing the lintian error still in https://salsa.debian.org/science-team/tango/-/jobs/808832 and also hardening-check returns the same.
- [ ] The database is not upgraded on package upgrade from 9.2.5 to 9.3.4.

I'm seeing

```
/usr/sbin/dbconfig-generate-include: 246: cannot create /tmp/dbconfig-generate-include.EBasHV: Permission denied
```

which looks related when doing

```
apt remove --yes --purge "*tango*" && rm -f /var/tmp/*mysql* && apt install --yes tango-db tango-starter tango-test && apt install --yes -t experimental tango-db tango-starter tango-test
```

I've verified that my method of checking the database upgrade with

```sql
USE tango;
SELECT * FROM property_class WHERE value ='EventConfirmSubscription';
```

works as I get something when I manually execute `update_db_from_9.2.5_to_9.3.4.sql`.

- [ ] (Just a comment): In the lintian log on salsa I see

```
$ if lintian --fail-on error --allow-root > /dev/null ; then # collapsed multi-line command
Unknown option: fail-on
```

General: @bourtemb @mliszcz 

- [ ] Logviewer is not working -> https://github.com/tango-controls/LogViewer/issues/5
- [x] DataBaseds ipv4 -> https://github.com/tango-controls/TangoTickets/issues/40

**Comment by picca** [comment]
Hello

here the dbconfig-common bug report about this Permission Denied

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=965219


I am also investigating about the hardening. It seems that the [X]FLAGS are not propagated into the Makefile.am.
It seems that it works for log4tango but not for all the tango sources.
I saw in some Makefile.am, some of these fortification harcoded. It seems to me that these flags should not be hardcoded in the Makefile.am but given during the configuration phase.

cheers

here the new build

https://salsa.debian.org/science-team/tango/-/jobs/954274/raw

**Comment by picca** [comment]
here the relevant part of the build

```
make[6]: Leaving directory '/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1/cppserver/AbstractClass'
make[5]: Leaving directory '/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1/cppserver/AbstractClass'
make[5]: Entering directory '/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1/cppserver/tangoaccesscontrol'
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../lib/cpp/client -I../../lib/cpp/server -I/usr/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/server -I../../cppserver/AbstractClass/AccessControl -I/usr/include/mysql -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include   -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o ClassFactory.o ClassFactory.cpp
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../lib/cpp/client -I../../lib/cpp/server -I/usr/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/server -I../../cppserver/AbstractClass/AccessControl -I/usr/include/mysql -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include   -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o TangoAccessControlClass.o TangoAccessControlClass.cpp
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../lib/cpp/client -I../../lib/cpp/server -I/usr/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/server -I../../cppserver/AbstractClass/AccessControl -I/usr/include/mysql -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include   -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o TangoAccessControl.o TangoAccessControl.cpp
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../lib/cpp/client -I../../lib/cpp/server -I/usr/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/server -I../../cppserver/AbstractClass/AccessControl -I/usr/include/mysql -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include   -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o TangoAccessControlStateMachine.o TangoAccessControlStateMachine.cpp
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../lib/cpp/client -I../../lib/cpp/server -I/usr/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/server -I../../cppserver/AbstractClass/AccessControl -I/usr/include/mysql -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include   -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o DbUtils.o DbUtils.cpp
g++ -DHAVE_CONFIG_H -I. -I../..  -I../../lib/cpp/client -I../../lib/cpp/server -I/usr/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/log4tango/include -I../../lib/cpp/server -I../../cppserver/AbstractClass/AccessControl -I/usr/include/mysql -isystem /usr/include/mit-krb5 -I/usr/include/pgm-5.2 -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include   -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -c -o main.o main.cpp
/bin/bash ../../libtool  --tag=CXX   --mode=link g++  -g -O2 -fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=. -fstack-protector-strong -Wformat -Werror=format-security   -D_REENTRANT -DOMNI_UNLOADABLE_STUBS  -Wl,-z,relro -Wl,-z,now -o TangoAccessControl ClassFactory.o TangoAccessControlClass.o TangoAccessControl.o TangoAccessControlStateMachine.o DbUtils.o main.o -L../../lib/cpp/client -ltango  -lmysqlclient -lzmq ../AbstractClass/AccessControl/libaccesscontrol.la -lpthread -lnsl -ldl -L/usr/lib/bfd-plugins -lomniORB4 -lomniDynamic4 -lCOS4  -lpthread -lomnithread -lnsl -L/usr/lib/bfd-plugins  -lpthread -lomnithread -lpthread
libtool: link: g++ -g -O2 "-fdebug-prefix-map=/builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1=." -fstack-protector-strong -Wformat -Werror=format-security -D_REENTRANT -DOMNI_UNLOADABLE_STUBS -Wl,-z -Wl,relro -Wl,-z -Wl,now -o .libs/TangoAccessControl ClassFactory.o TangoAccessControlClass.o TangoAccessControl.o TangoAccessControlStateMachine.o DbUtils.o main.o  -L../../lib/cpp/client /builds/science-team/tango/debian/output/tango-9.3.4~rc6+dfsg1/lib/cpp/client/.libs/libtango.so -lmysqlclient -lzmq ../AbstractClass/AccessControl/.libs/libaccesscontrol.a -L/usr/lib/bfd-plugins -ldl -lomniORB4 -lomniDynamic4 -lCOS4 -lnsl -lomnithread -lpthread
```

so it seems that some of the hardening flags are used

**Comment by picca** [comment]
Here the official flags

```
picca@sixs3:~/debian/science-team/tango$ dpkg-buildflags 
CFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong -Wformat -Werror=format-security
CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2
CXXFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong -Wformat -Werror=format-security
FCFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong
FFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong
GCJFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong
LDFLAGS=-Wl,-z,relro
OBJCFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong -Wformat -Werror=format-security
OBJCXXFLAGS=-g -O2 -fdebug-prefix-map=/home/experiences/instrumentation/picca/debian/science-team/tango=. -fstack-protector-strong -Wformat -Werror=format-security
```

**Comment by picca** [comment]
here the output of hardening-check for the database and the TangoAcessControl
```
picca@sixs3:~$ hardening-check -v TangoAccessControl 
TangoAccessControl:
 Position Independent Executable: yes
 Stack protected: yes
 Fortify Source functions: no, only unprotected functions found!
	unprotected: memcpy
	unprotected: memmove
	unprotected: strcpy
	unprotected: gethostname
 Read-only relocations: yes
 Immediate binding: yes
 Stack clash protection: unknown, no -fstack-clash-protection instructions found
 Control flow integrity: no, not found!
picca@sixs3:~$ hardening-check -v DataBaseds 
DataBaseds:
 Position Independent Executable: yes
 Stack protected: yes
 Fortify Source functions: yes (some protected functions found)
	unprotected: strcpy
	unprotected: stpcpy
	unprotected: memcpy
	unprotected: gethostname
	unprotected: memmove
	protected: strcpy
	protected: strcat
	protected: sprintf
 Read-only relocations: yes
 Immediate binding: yes
 Stack clash protection: unknown, no -fstack-clash-protection instructions found
 Control flow integrity: no, not found!
```

**Comment by picca** [comment]
Re-hello,

I find this in the code of the device.

```
void TangoAccessControl::init_device()
{
	DEBUG_STREAM << "TangoAccessControl::init_device() create device " << device_name << endl;
	/*----- PROTECTED REGION ID(TangoAccessControl::init_device_before) ENABLED START -----*/
	
	//	Initialization before get_device_property() call
	
	/*----- PROTECTED REGION END -----*/	//	TangoAccessControl::init_device_before
	
	if (Tango::Util::instance()->is_svr_starting() == false  &&
		Tango::Util::instance()->is_device_restarting(device_name)==false)
	{
		//	If not starting up call init device for inherited object
		AccessControl_ns::AccessControl::init_device();
	}
	//	No device property to be read from database
	

	/*----- PROTECTED REGION ID(TangoAccessControl::init_device) ENABLED START -----*/

	// These lines have obsolutely no other goals than to make the
	// hardening_check command happy. Without them, this command will return
	// missing stack-protection and missing "Fortify source"
	// This is for Debian packages hardening effort

	char host[256];
	gethostname(host,sizeof(host));
	sprintf(host,"Hello world\n");

	//	Initialize device
	// Initialise variables to default values
	//--------------------------------------------

	string str_rcs(RcsId);

	mysql_connection();

	set_state(Tango::ON);
	set_status("Device is OK");

	/*----- PROTECTED REGION END -----*/	//	TangoAccessControl::init_device
}
```

Is the Debian specifique code still relevant ?

I do not like the idea to add some random crap in a Device in order to silent a check which can also trigger faulse positiv.

cheers

**Comment by picca** [comment]
Once removed I have this.

```
picca@sixs3:~$ hardening-check -v TangoAccessControl 
TangoAccessControl:
 Position Independent Executable: yes
 Stack protected: yes
 Fortify Source functions: no, only unprotected functions found!
	unprotected: memcpy
	unprotected: strcpy
	unprotected: memmove
 Read-only relocations: yes
 Immediate binding: yes
 Stack clash protection: unknown, no -fstack-clash-protection instructions found
 Control flow integrity: no, not found!
```

no more gethostname in the list of unprotected functions

**Comment by t-b** [comment]
@picca 

> Is the Debian specifique code still relevant ?
>
> I do not like the idea to add some random crap in a Device in order to silent a check which can also trigger faulse > positiv.

Me too, I've opened https://github.com/tango-controls/TangoAccessControl/pull/4 to remove that.

Regarding the hardening flags, looks like a false positive indeed. I did some more digging and all the flags are passed correctly. Yes the build system is indeed rather ancient. This is planned to get better once we've switched to cmake for everything.

> here the dbconfig-common bug report about this Permission Denied
> https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=965219

Thanks for digging that up. I can confirm that this is only a bug when upgrading the tango database. On a fresh install, without prior tango database, this works. Is there a workaround available to unbreak the database upgrade?

We are currently preparing rc7 which hopefully is also the last rc.

**Comment by t-b** [comment]
@picca I've created rc7. Would be nice to have new debian packages with the database permission issue fixed. Thanks.

**Comment by picca** [comment]
I can upload the new package but, I would like to solve the problem at the dbconfig level.

Is it possible for you to provide the full log of the upgrade issue.
This way I will be able to discuss with the dbconfig upstream.
the best would be to add this log informations into the dbconfig-common bug report

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=965219

cheers

Fred

**Comment by t-b** [comment]
@picca Will do. I'll ping you when it's done. Is a simple shell copy and paste enough or should I enable some special debug mode?

**Comment by picca** [comment]
I just uploaded rc7 into Debian experimental, but there is no fix for the upgrade. We will try to deal with this before the final release.

Do you know if the java part works out of the box with openjdk11 ?

**Comment by t-b** [comment]
> Do you know if the java part works out of the box with openjdk11 ?

We still require java 8 for the java part.

**Comment by t-b** [comment]
> Is it possible for you to provide the full log of the upgrade issue.
This way I will be able to discuss with the dbconfig upstream.
the best would be to add this log informations into the dbconfig-common bug report

Done.

**Comment by picca** [comment]
Thanks
________________________________________
De : Thomas Braun [notifications@github.com]
Envoyé : jeudi 10 septembre 2020 12:59
À : tango-controls/TangoSourceDistribution
Cc : PICCA Frederic-Emmanuel; Mention
Objet : Re: [tango-controls/TangoSourceDistribution] Feedback testing debian packages: 9.3.4-rcx (#67)

Is it possible for you to provide the full log of the upgrade issue.
This way I will be able to discuss with the dbconfig upstream.
the best would be to add this log informations into the dbconfig-common bug report

Done.

**Comment by t-b** [comment]
Closing as we now a final 9.3.4 release, see I've updated the release notes for the TDS, see https://github.com/tango-controls/TangoSourceDistribution/releases/tag/9.3.4.

**Comment by t-b** [comment]
@picca I did some more debian package testing as requested.

- [ ] Database upgrade issue

Installation of new packages without old packages:

```
root@debian:~# apt remove --purge "*maria*" "*mysql*" "*tango*" --yes && rm /var/tmp/tango-db*; apt install mariadb-server tango-db -t experimental --yes
```

results in the correct database being installed.

Installation of new packages with old packages present:

```
apt remove --purge "*maria*" "*mysql*" "*tango*" --yes && rm /var/tmp/tango-db*; apt install tango-db mariadb-server --yes && apt install tango-db mariadb-server -t experimental --yes
```

results in the database not being upgraded. The dbconfig-common error is gone though.

I'm using

```
mysql -u --password
USE tango;
SELECT * FROM property_class WHERE value ='EventConfirmSubscription';
```

to check the database. This gives an empty set for the 9.2.5 database and a non-empty set for 9.3.4 database.

- [ ] Installation order issue

Using the following sequence

```
apt remove --purge "*maria*" "*mysql*" "*tango*" --yes && rm /var/tmp/tango-db*; apt install mariadb-server --yes -t experimental && apt install tango-db -t experimental --yes
```

I don't get a working installation as the database is not reachable at `/var/run/mysqld/mysqld.sock`.
